chain/network/src/peer_manager/tests/snapshot_hosts.rs

```diff
@@ -174,7 +174,7 @@ async fn invalid_signature_not_broadcast() {
     let ok_info_b = make_snapshot_host_info(&peer1_config.node_id(), &peer1_config.node_key, rng);
 
     let invalid_message = PeerMessage::SyncSnapshotHosts(SyncSnapshotHosts {
-        hosts: vec![ok_info_a, invalid_info, ok_info_b],
+        hosts: vec![ok_info_a.clone(), invalid_info, ok_info_b.clone()],
     });
     peer1.send(invalid_message).await;
```

Comment:
I didn't add a check here because I can't wait for any sort of event. After peer1 sends the invalid message the state of PeerManager can just be empty and there could be no events at all. I could verify that the state doesn't contain invalid_info, but there's uncertainty - there's a chance that I would check the state before invalid_info is put into it.

Overall I don't think that putting an uncertain check here would give any big benefit, the check at the end is enough.